src/render_tasks/d3d12_raytracing_task.hpp

```diff
@@ -144,34 +155,77 @@ namespace wr
 			auto& data = fg.GetData<RaytracingData>(handle);
 			auto& as_build_data = fg.GetPredecessorData<wr::ASBuildData>();
 
+			d3d12::DescriptorHeap* desc_heap = cmd_list->m_rt_descriptor_heap->GetHeap();
+
 			cmd_list->m_native->ResourceBarrier(1, &CD3DX12_RESOURCE_BARRIER::UAV(as_build_data.out_tlas.m_native));
 
 			if (n_render_system.m_render_window.has_value())
 			{
 				auto frame_idx = n_render_system.GetFrameIdx();
 
+				d3d12::BindRaytracingPipeline(cmd_list, data.out_state_object, d3d12::GetRaytracingType(device) == RaytracingType::FALLBACK);
+
+				d3d12::SetRTShaderUAV(cmd_list, 0, COMPILATION_EVAL(rs_layout::GetHeapLoc(params::full_raytracing, params::FullRaytracingE::OUTPUT)), data.out_uav_from_rtv.GetDescriptorHandle());
+				d3d12::SetRTShaderSRV(cmd_list, 0, COMPILATION_EVAL(rs_layout::GetHeapLoc(params::full_raytracing, params::FullRaytracingE::INDICES)), as_build_data.out_scene_ib_alloc.GetDescriptorHandle());
+				d3d12::SetRTShaderSRV(cmd_list, 0, COMPILATION_EVAL(rs_layout::GetHeapLoc(params::full_raytracing, params::FullRaytracingE::MATERIALS)), as_build_data.out_scene_mat_alloc.GetDescriptorHandle());
+				d3d12::SetRTShaderSRV(cmd_list, 0, COMPILATION_EVAL(rs_layout::GetHeapLoc(params::full_raytracing, params::FullRaytracingE::OFFSETS)), as_build_data.out_scene_offset_alloc.GetDescriptorHandle());
+
+				//Test, bind all textures. This will be changed
+				auto hndl = as_build_data.out_material_handles[0];
+				auto mat_int = hndl->m_pool->GetMaterial(hndl->m_id);
+				auto* txture = static_cast<wr::d3d12::TextureResource*>(mat_int->GetAlbedo().m_pool->GetTexture(mat_int->GetAlbedo().m_id));
+
```

Comment:
the following loop will be replaced by some other strategy. Maybe I can initialize the heap class with dummy descriptors so that the copydescriptor function doesn't break.